### PR TITLE
Simplified MANIFEST.in

### PR DESCRIPTION
**Pull request by timgraham**
Also fixed missing django/contrib/auth/common-passwords.txt.gz file and added JavaScript tests.

mailing list discussion: https://groups.google.com/d/topic/django-developers/d6UWv53MwhE/discussion

Output of check-manifest:
```
lists of files in version control and sdist do not match!
missing from sdist:
  .editorconfig
  .eslintignore
  .eslintrc
  .gitattributes
  .tx
  .tx/config
you have django/conf/locale/af/LC_MESSAGES/django.mo in source control!
that's a bad idea: auto-generated files should not be versioned
this also applies to the following:
  django/conf/locale/ar/LC_MESSAGES/django.mo
  .... (all mo files)
```

### CODE REVIEW

File: MANIFEST.in [hunk 1]
```diff
@@ -1,26 +1,16 @@
-include README.rst
```
Comment (MoritzS) [inline MANIFEST.in]: Why did you remove the README?

Comment (timgraham) [inline MANIFEST.in]: Replaced with `include *.rst`
